src/atoms/Switch/Switch.native.js

```diff
@@ -189,6 +189,20 @@ const Switch = ({ disabled, on, defaultOn, onChange, size, testID }) => {
     new Animated.Value(animationConfig.off.rightSpace),
   );
 
+  useEffect(() => {
+    if (defaultOn) {
```

Comment:
why was this required?

Reply:
This was for `defaultOn`. You did the eslint formating thing, which introduced the bug as I hadn't added all the dependencies in the useEffect.